src/magic.rs

```diff
@@ -41,43 +41,39 @@ struct MagicRule {
 
 impl MagicRule {
     fn matches_data(&self, data: &[u8]) -> bool {
+        // Do we need the value_length at all, if it's implicit in
+        // value.len() and checked by the parsers?
```

Comment:
That's a fair question, and I think the answer is: no, we don't need to store the length at all.

Reply:
I've removed the comment but kept the assertions.  Per the commit message,

```
Remove comment that wonders whether checking value/mask sizes is necessary

It is.  Right now it is done in the parser, but hopefully we can move
to model where a) the parser enforces that; b) a MagicRule::new() or
MagicRuleBuilder asserts that.
```